lib/record_list.rb

```diff
@@ -0,0 +1,32 @@
+require_relative 'entry'
+
+class RecordList
+
+  attr_reader :list, :balance
+
+  def create_entry(amount, entry = Entry.new)
+    update_balance(amount)
+    entry.set_attributes(amount, @balance)
+    add_to_list(entry)
+  end
+
+  def get_list
+    @list
+  end
+
+  private
+
+  def add_to_list(entry)
+    @list.push(entry)
+  end
+
+  def initialize
+    @list = []
+    @balance = 0
```

Comment:
Is it possible to refactor the balance code so that balance is only calculated when needed, rather than a running total being kept?